### PR TITLE
Issue#2760  converted data source.java

### LINKED ISSUE #2760: Convert Datasource.java to Datasource.kt
parent issue #1826

What changes you will do:
will convert Java file to Kotlin

I couldn't find an issue regarding this, do tell me if there is I will close this

### PR DESCRIPTION
**Pull request by cdhiraj40**
fixes #2760

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModel.kt [hunk 1]
```diff
@@ -462,15 +462,16 @@ class ZimManageViewModel @Inject constructor(
...
-          )
-        },
-        Throwable::printStackTrace
-      )
+      .subscribe({ newList ->
+        fileSelectListStates.postValue(
+          fileSelectListStates.value?.let {
+            inheritSelections(
+              it,
+              newList as MutableList<BooksOnDiskListItem>
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModel.kt:470]: We can call it as `newList.toMutableList()`.

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/Repository.kt [hunk 2]
```diff
@@ -114,7 +114,7 @@ class Repository @Inject internal constructor(
     Completable.fromAction { bookmarksDao.deleteBookmarks(bookmarks) }
       .subscribeOn(io)
 
-  override fun deleteBookmark(bookmarkUrl: String): Completable? =
+  override fun deleteBookmark(bookmarkUrl: String): Completable =
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/Repository.kt:117]: Make `Completable` null here. We had an issue here.

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/MainRepositoryActions.kt [hunk 3]
```diff
@@ -41,6 +41,6 @@ class MainRepositoryActions @Inject constructor(private val dataSource: DataSour
 
   fun deleteBookmark(bookmarkUrl: String) {
     dataSource.deleteBookmark(bookmarkUrl)
-      .subscribe({}, { e -> Log.e(TAG, "Unable to delete bookmark", e) })
+      ?.subscribe({}, { e -> Log.e(TAG, "Unable to delete bookmark", e) })
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/MainRepositoryActions.kt:44]: We should log also if the database fails.

**Comment by gouri-panda** [comment]
@cdhiraj40 Any update?

**Comment by cdhiraj40** [comment]
> @cdhiraj40 Any update?

sorry for the late reply, I was busy with my academic studies.
I will update the PR soon.

**Comment by gouri-panda** [comment]
@cdhiraj40 It's Ok :)

**Comment by kelson42** [comment]
@cdhiraj40 @gouri-panda What shoukd we do with this PR?

**Comment by gouri-panda** [comment]
@kelson42  It's just interface class and some refactoring. It shouldn't take long to review.

**Review by gouri-panda — APPROVED** [review]

**Pull request merged**